reactive/openvim.py

```diff
@@ -202,21 +202,22 @@ def send_ssh_key(compute):
 @when('compute.available', 'openvim-controller.installed')
 def host_add(compute):
     cache = kv()
-    if cache.get("compute:" + compute.address()):
-        return
-    #TODO See if there's a charm helper for this.
-    cmd = "ssh -n -o 'StrictHostKeyChecking no' %s@%s"
-    sh_as_openvim(cmd % (compute.user(), compute.address()))
-    data = {
-        'host': {
-            'name': 'compute-0',
-            'user': compute.user(),
-            'ip_name': compute.address(),
-            'description': 'compute-0'
+    for node in compute.authorized_nodes():
+        if cache.get("compute:" + node['address']):
+            continue
+        #TODO See if there's a charm helper for this.
+        cmd = "ssh -n -o 'StrictHostKeyChecking no' %s@%s"
+        sh_as_openvim(cmd % (node['user'], node['address']))
+        data = {
+            'host': {
+                'name': 'compute-0',
```

Comment:
you absolutely should, and it should be based on JUJU_REMOTE_UNIT which you can get as conversation scope name